### DISCUSSION TITLE
'AADSTS700016: Application with identifier '31359c7f-bd7e-475c-86db-fdb8c937548e' was not found in the directory

### DISCUSSION THREAD

**Discussion by maikbergerhoff** [pnp/pnpcore#1529] [Q&A] [answered]
Since yesterday, I have been receiving the following error message when attempting to start a login by using Username Password Authentication:

“Microsoft.Identity.Client.MsalServiceException: 'AADSTS700016: Application with identifier '31359c7f-bd7e-475c-86db-fdb8c937548e' was not found in the directory..This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You may have sent your authentication request to the wrong tenant”

We check the configs, but all values are set correct…
The account we us is excluded from MFA-Policy. We are using a console app with the following code:

using AnalyseSPItemBySearch.Logic;
using AnalyseSPItemBySearch.Models;
using AnalyseSPItemBySearch.Repo;
using IKB.Api.Common.Models;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using Microsoft.Extensions.Logging;
using PnP.Core.Auth.Services.Builder.Configuration;
using PnP.Core.Services;
using PnP.Core.Services.Builder.Configuration;
using System.Reflection;
using System.Security.Cryptography.X509Certificates;

namespace AnalyseSPItemBySearch
{
    class Program
    {
        public static async Task Main(string[] args)
        {
            ////Read App settings
            IConfigurationRoot config = new ConfigurationBuilder()
                  .SetBasePath(Directory.GetCurrentDirectory())
                  .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
                  .AddUserSecrets(Assembly.GetExecutingAssembly(), true)
                  .Build();

            // Creates and configures the host
            IHost host = Host.CreateDefaultBuilder()
                .ConfigureLogging((hostingContext, logging) =>
                {
                    logging.AddEventSourceLogger();
                    logging.AddConsole();
                })
                .ConfigureServices((hostingContext, services) =>
                {
                    // Read the custom configuration from the appsettings.<environment>.json file
                    CustomSettings customSettings = new();
                    hostingContext.Configuration.Bind("CustomSettings", customSettings);

                    // Add PnP Core SDK
                    services.AddPnPCore(options =>
                    {
                        options.PnPContext.GraphFirst = true;
                        options.HttpRequests.UserAgent = "NONISV|SharePointPnP|PnPCoreSDK";
                        options.HttpRequests.MicrosoftGraph = new PnPCoreHttpRequestsGraphOptions
                        {
                            UseRetryAfterHeader = true,
                            MaxRetries = 10,
                            DelayInSeconds = 3,
                            UseIncrementalDelay = true,
                        };
                        options.HttpRequests.SharePointRest = new PnPCoreHttpRequestsSharePointRestOptions
                        {
                            UseRetryAfterHeader = true,
                            MaxRetries = 10,
                            DelayInSeconds = 3,
                            UseIncrementalDelay = true,
                        };
                        options.Sites.Add("Site",
                            new PnP.Core.Services.Builder.Configuration.PnPCoreSiteOptions
                            {
                                SiteUrl = customSettings.SearchSiteUrl
                            });
                    });

                    // PnP Core Authentication
                    // To check out more authentication options check out the documentation for more information:
                    //  https://pnp.github.io/pnpcore/using-the-sdk/configuring%20authentication.html
                    services.AddPnPCoreAuthentication(options =>
                    {

                        ////Username & Password

                        options.Credentials.Configurations.Add("UsernamePassword",
                          new PnPCoreAuthenticationCredentialConfigurationOptions
                          {

                              ClientId = config["ClientId"],
                              TenantId = config["TenantId"],
                              UsernamePassword = new PnPCoreAuthenticationUsernamePasswordOptions
                              {
                                  Username = config["SPAdmin"],
                                  Password = config["SPAdminPassword"]
                              }
                          });

                        options.Credentials.DefaultConfiguration = "UsernamePassword"; // Set default configuration

                        options.Sites.Add("Site", new PnP.Core.Auth.Services.Builder.Configuration.PnPCoreAuthenticationSiteOptions
                        {
                            AuthenticationProviderName = "UsernamePassword"
                        });

                    });
                })
                // Let the builder know we're running in a console
                .UseConsoleLifetime()
                // Add services to the container
                .Build();

            // Start the host
            await host.StartAsync();

            // create logger
            var serviceProvider = new ServiceCollection()
              .AddLogging(builder => builder.AddConsole())
              .BuildServiceProvider();
            var logger = serviceProvider.GetService<ILogger<Program>>();
           
            
            using (var scope = host.Services.CreateScope())
            {
                var pnpContextFactory = scope.ServiceProvider.GetRequiredService<IPnPContextFactory>();
                using (var context = await pnpContextFactory.CreateAsync(new Uri("https://xyz.sharepoint.com/sites/test")))
                {
                    var web = await context.Web.GetAsync();
                    Console.WriteLine($"Web title: {web.Title}");
                }
            }

        }
    }
}

I would be very grateful for any help.

**Comment by pulipatiram** [comment, marked as answer]
This issue occurred because of security change in Pnp: Due to which your code started to fail with error: 
[https://pnp.github.io/blog/post/changes-pnp-management-shell-registration/](https://urldefense.com/v3/__https:/pnp.github.io/blog/post/changes-pnp-management-shell-registration/__;!!MBaoSxY!54h0SAyXvQNPy70uDwLXnm99E90zOr1GnbCV1TBUfiRqDTC6f-Y7xfYAP7oWwFVmgYFB03iClhA2kxBFh9btqcWIcA$)  where we do not allow multi tenant pnp app.
-	Register new App in Azure AD for single with tenant configuration
-	Use REST calls for which we updated the code to create instance of  “AuthenticationManager.CreateWithCertificate” as shown below.
- Ref: https://learn.microsoft.com/en-us/sharepoint/dev/sp-add-ins-modernize/from-acs-to-aad-apps?source=recommendations#consuming-sharepoint-online-in-app-only-mode-via-an-azure-ad-registered-application
Hope this helps.. Mark it as answer if problem is resolved.

**Reply by jansenbe** [reply]
Thanks for the details @pulipatiram . Yesterday I've also removed all references and uses of the PnP Management Shell app from PnP Core SDK and PnP Framework.

**Comment by maikbergerhoff** [comment]
Thanks for Support!!!
